src/Workspaces/Core/Portable/Shared/Extensions/ISolutionExtensions.cs

```diff
@@ -48,19 +50,30 @@ public static IEnumerable<DocumentId> GetChangedDocuments(this Solution newSolut
 
         public static TextDocument GetTextDocument(this Solution solution, DocumentId documentId)
         {
-            return solution.GetDocument(documentId) ?? solution.GetAdditionalDocument(documentId);
+            return solution.GetDocument(documentId) ?? solution.GetAdditionalDocument(documentId) ?? solution.GetAnalyzerConfigDocument(documentId);
+        }
+
+        public static TextDocumentKind GetDocumentKind(this Solution solution, DocumentId documentId)
+        {
+            return solution.GetTextDocument(documentId).Kind;
         }
 
         public static Solution WithTextDocumentText(this Solution solution, DocumentId documentId, SourceText text, PreservationMode mode = PreservationMode.PreserveIdentity)
         {
-            var document = solution.GetTextDocument(documentId);
-            if (document is Document)
-            {
-                return solution.WithDocumentText(documentId, text, mode);
-            }
-            else
+            var documentKind = solution.GetDocumentKind(documentId);
+            switch (documentKind)
```

Comment:
I see, you have one for text changes. then, probably want for all other ones as well?